TESTS/mbed_hal/qspi/flash_configs/N25Q128A_config.h

```diff
@@ -0,0 +1,226 @@
+#define CONFIG2_BIT_ODS1   (1 << 1)   // Output driver strength
+#define CONFIG2_BIT_ODS2   (1 << 2)   // Output driver strength
+#define CONFIG2_BIT_VPP    (1 << 3)   // VPP accelerator            1 = Disabled (Default)
+#define CONFIG2_BIT_RH     (1 << 4)   // Reset/hold
+#define CONFIG2_BIT_DE     (1 << 6)   // Dual I/O protocol          0 = Enabled / 1 = Disabled (Default, extended SPI protocol)
+#define CONFIG2_BIT_QE     (1 << 7)   // Quad I/O protocol          0 = Enabled / 1 = Disabled (Default, extended SPI protocol)
+#define CONFIG2_BITS_DEFAULT    0xDF  // reg default state
+
+
+#define DUAL_ENABLE_IMPLEMENTATION()                                            \
+                                                                                \
+    uint8_t reg_data[QSPI_CONFIG_REG_2_SIZE];                                   \
+                                                                                \
+    memset(reg_data, 0, QSPI_CONFIG_REG_2_SIZE);                                \
+    if (read_register(QSPI_CMD_RDCR2, reg_data,                                 \
+            QSPI_CONFIG_REG_2_SIZE, qspi) != QSPI_STATUS_OK) {                  \
+        return QSPI_STATUS_ERROR;                                               \
+    }                                                                           \
+    if (write_enable(qspi) != QSPI_STATUS_OK) {                                 \
+        return QSPI_STATUS_ERROR;                                               \
+    }                                                                           \
+    WAIT_FOR(WRSR_MAX_TIME, qspi);                                              \
+                                                                                \
+    reg_data[0] = reg_data[0] & ~(CONFIG2_BIT_DE);                              \
+    if (write_register(QSPI_CMD_WRCR2, reg_data,                                \
+            QSPI_CONFIG_REG_2_SIZE, qspi) != QSPI_STATUS_OK) {                  \
+        return QSPI_STATUS_ERROR;                                               \
+    }                                                                           \
+    WAIT_FOR(WRSR_MAX_TIME, qspi);                                              \
+    memset(reg_data, 0, QSPI_CONFIG_REG_2_SIZE);                                \
+    if (read_register(QSPI_CMD_RDCR2, reg_data,                                 \
+            QSPI_CONFIG_REG_2_SIZE, qspi) != QSPI_STATUS_OK) {                  \
+        return QSPI_STATUS_ERROR;                                               \
+    }                                                                           \
+                                                                                \
+    return ((reg_data[0] & CONFIG2_BIT_DE) == 0 ?                               \
+                QSPI_STATUS_OK : QSPI_STATUS_ERROR)
+
+
+#define DUAL_DISABLE_IMPLEMENTATION()                                           \
```

Comment:
why these are not inlined functions rather than large macro function definitions?

Reply:
no specific reasons, can change macros to static inline functions